aea/test_tools/test_cases.py

```diff
@@ -97,6 +99,11 @@
 class BaseAEATestCase(ABC):  # pylint: disable=too-many-public-methods
     """Base class for AEA test cases."""
 
+    @pytest.fixture(autouse=True)
+    def set_capfd_on_cli_runner(self, capfd: CaptureFixture) -> None:
+        """Set pytest capfd on CLI runner"""
+        self.runner.capfd = capfd
```

Comment:
I don't think this is a good idea, this make the `pytest` a dependency for `BaseAEATestCase` and why are we using the `capfd` fixture here? 

Also why not make this fixture to be scoped at function level? Since if this is class level won't it effect the output capture for different tests in the same class?

Reply:
- this is true, any test in our code base using a fixture depends on pytest. Most of our tests are also written in pytest-style, that is with `setup_class`, `teardown_class` etc. and not inheriting from `unittest.TestCase`
- a fixture is (by default) scoped at function level. This one is as well.